### PR TITLE
Support K-LMS in img2img

### PR DESCRIPTION
**Pull request by anton-l**
While looking into https://github.com/huggingface/diffusers/pull/253 I've discovered that our img2img script is incompatible with K-LMS for two reasons:
* `lms_scheduler.timesteps` are not the same as `ddim_scheduler.timesteps`: they are fractional (`torch.float`) when they are passed to `unet.forward()`, and they need to be `enumerate()`'d  to be used with `scheduler.step(timestep=i, ...)`.
* the model inputs need to be scaled by `1 / ((sigma**2 + 1) ** 0.5)` to match the ODE formulation

This PR fixes the issues and is compatible with https://github.com/huggingface/diffusers/pull/253 

#253 is to be merged before this PR!

### CODE REVIEW

File: examples/inference/image_to_image.py [hunk 1]
```diff
@@ -135,6 +147,11 @@ def __call__(
         for i, t in tqdm(enumerate(self.scheduler.timesteps[t_start:])):
             # expand the latents if we are doing classifier free guidance
             latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
+            if isinstance(self.scheduler, LMSDiscreteScheduler):
+                sigma = self.scheduler.sigmas[t_start + i]
+                latent_model_input = latent_model_input / ((sigma**2 + 1) ** 0.5)
+                latent_model_input = latent_model_input.to(self.unet.dtype)
+                t = t.to(self.unet.dtype)
```
Comment (anton-l) [inline examples/inference/image_to_image.py:154]: For some reason `autocast` stopped working for these, so I'm adding redundant type casts. Does someone have an intuition for why that is?

Comment (patrickvonplaten) [reply examples/inference/image_to_image.py:154]: Does it work now with `autocast` or not?

Comment (anton-l) [reply examples/inference/image_to_image.py:154]: This version works with autocast, yes

File: src/diffusers/schedulers/scheduling_lms_discrete.py [hunk 2]
```diff
@@ -124,10 +124,9 @@ def step(
         return {"prev_sample": prev_sample}
 
     def add_noise(self, original_samples, noise, timesteps):
-        alpha_prod = self.alphas_cumprod[timesteps]
-        alpha_prod = self.match_shape(alpha_prod, original_samples)
+        sigmas = self.match_shape(self.sigmas[timesteps], noise)
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_lms_discrete.py:127]: We should give the original author credit here and instead merge: https://github.com/huggingface/diffusers/pull/253

Comment (anton-l) [reply src/diffusers/schedulers/scheduling_lms_discrete.py:127]: Of course, the plan is to merge the original PR before this :)

File: examples/inference/image_to_image.py [hunk 3]
```diff
@@ -135,6 +147,11 @@ def __call__(
         for i, t in tqdm(enumerate(self.scheduler.timesteps[t_start:])):
             # expand the latents if we are doing classifier free guidance
             latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
+            if isinstance(self.scheduler, LMSDiscreteScheduler):
+                sigma = self.scheduler.sigmas[t_start + i]
+                latent_model_input = latent_model_input / ((sigma**2 + 1) ** 0.5)
```
Comment (patrickvonplaten) [inline examples/inference/image_to_image.py:152]:
```suggestion
                # something like "cast latent model input into "sigma" space ...  => think you know better how to describe it :-)
                latent_model_input = latent_model_input / ((sigma**2 + 1) ** 0.5)
```

File: examples/inference/image_to_image.py [hunk 4]
```diff
@@ -135,6 +147,11 @@ def __call__(
         for i, t in tqdm(enumerate(self.scheduler.timesteps[t_start:])):
```
Comment (patrickvonplaten) [inline examples/inference/image_to_image.py:147]: Even though `i` is only used for LMS, I think it could be a bit easier to understand if we add a variable like:
```suggestion
        for i, t in tqdm(enumerate(self.scheduler.timesteps[t_start:])):
            timestep_index = t_start + i
```

Comment (patrickvonplaten) [reply examples/inference/image_to_image.py:147]: or just `t_index`

Comment (patrickvonplaten) [reply examples/inference/image_to_image.py:147]: And then just use this instead of `t_start + i` in the sigmas. IMO this could help a bit the readability here
